### ISSUE TITLE
Random tensors are not reproducible with global seed set

### ISSUE THREAD

**Issue by merajhashemi** [tensorly/tensorly#208]
Currently setting numpy.seed, torch.manual_seed, etc., is not sufficient for reproducible results when a seed is not explicitly set for the random tensor functions. 
Inspired by scipy and scikit-learn`check_random_state` function, 
https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/utils/validation.py#L862
https://github.com/scipy/scipy/blob/624fd766e2886c458012d98e3c6ec85551ab2916/scipy/_lib/_util.py#L201
we can change the `check_random_state` function to return the global numpy RNG if the seed is `None`. This will make sure setting `np.random.seed(seed)` is enough for reproducible results.
